integration/internal/api/system/system.go

```diff
@@ -0,0 +1,45 @@
+package system
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/internal/test/environment"
+	"github.com/stretchr/testify/require"
+)
+
+// Time provides the current time on the daemon host
+func Time(t *testing.T, client client.APIClient, testEnv *environment.Execution) time.Time {
+	if testEnv.IsLocalDaemon() {
+		return time.Now()
+	}
+
+	ctx := context.Background()
+	info, err := client.Info(ctx)
+	require.Nil(t, err)
+
+	dt, err := time.Parse(time.RFC3339Nano, info.SystemTime)
+	require.Nil(t, err, "invalid time format in GET /info response")
+	return dt
+}
+
+// Version provides the version of the daemon
+func Version(client client.APIClient) (types.Version, error) {
```

Comment:
Please delete, this doesn't need to exist, just call `client.ServerVersion()` directly.

Reply:
This exists to provide a homogeneous CLI replacement interface.

Reply:
This is exactly the problem. We should not be testing as if we had a CLI. We need to test the API as directly as possible, not mimic CLI behaviour.

Reply:
As explained in my PR comment, that is a laudable possible goal for the future but these tests were written against the CLI.

Reply:
I guess I don't understand why we're moving these tests to `integration/` if they are going to be nearly identical to the `integration-cli` tests

Reply:
My understanding is that we're *moving* the tests to eliminate the `moby/moby` ➡️ `docker/cli` dependency and unblock additional test contributions as `integration-cli/` is now deprecated.